src/rgw/rgw_admin.cc

```diff
@@ -1061,2264 +67,371 @@ class StoreDestructor {
-{
-  bufferlist bl;
-  int ret = read_input(infile, bl);
-  if (ret < 0) {
-    cerr << "ERROR: failed to read input: " << cpp_strerror(-ret) << std::endl;
-    return ret;
-  }
-  JSONParser p;
-  if (!p.parse(bl.c_str(), bl.length())) {
-    cout << "failed to parse JSON" << std::endl;
-    return -EINVAL;
-  }
+  int64_t max_objects = -1;
+  int64_t max_size = -1;
+  bool have_max_objects = false;
+  bool have_max_size = false;
+  int include_all = false;
 
-  try {
-    t.decode_json(&p, k);
-  } catch (JSONDecoder::err& e) {
-    cout << "failed to decode JSON input: " << e.message << std::endl;
-    return -EINVAL;
-  }
-  return 0;
-}
+  int sync_stats = false;
+  int bypass_gc = false;
+  int warnings_only = false;
+  int inconsistent_index = false;
 
-static int parse_date_str(const string& date_str, utime_t& ut)
-{
-  uint64_t epoch = 0;
-  uint64_t nsec = 0;
+  int verbose = false;
 
-  if (!date_str.empty()) {
-    int ret = utime_t::parse_date(date_str, &epoch, &nsec);
-    if (ret < 0) {
-      cerr << "ERROR: failed to parse date: " << date_str << std::endl;
-      return -EINVAL;
-    }
-  }
+  int extra_info = false;
 
-  ut = utime_t(epoch, nsec);
+  uint64_t min_rewrite_size = 4 * 1024 * 1024;
+  uint64_t max_rewrite_size = ULLONG_MAX;
+  uint64_t min_rewrite_stripe_size = 0;
 
-  return 0;
-}
+  BIIndexType bi_index_type = PlainIdx;
 
-template <class T>
-static bool decode_dump(const char *field_name, bufferlist& bl, Formatter *f)
-{
-  T t;
+  string job_id;
+  int num_shards = 0;
+  bool num_shards_specified = false;
+  int max_concurrent_ios = 32;
+  uint64_t orphan_stale_secs = (24 * 3600);
 
-  bufferlist::iterator iter = bl.begin();
+  string err;
 
-  try {
-    decode(t, iter);
-  } catch (buffer::error& err) {
-    return false;
-  }
+  string source_zone_name;
+  string source_zone; /* zone id */
 
-  encode_json(field_name, t, f);
+  string tier_type;
+  bool tier_type_specified = false;
 
-  return true;
-}
+  map<string, string, ltstr_nocase> tier_config_add;
+  map<string, string, ltstr_nocase> tier_config_rm;
 
-static bool dump_string(const char *field_name, bufferlist& bl, Formatter *f)
-{
-  string val;
-  if (bl.length() > 0) {
-    val.assign(bl.c_str());
-  }
-  f->dump_string(field_name, val);
+  boost::optional<string> index_pool;
+  boost::optional<string> data_pool;
+  boost::optional<string> data_extra_pool;
+  RGWBucketIndexType placement_index_type = RGWBIType_Normal;
+  bool index_type_specified = false;
 
-  return true;
-}
+  boost::optional<std::string> compression_type;
 
-void set_quota_info(RGWQuotaInfo& quota, int opt_cmd, int64_t max_size, int64_t max_objects,
-                    bool have_max_size, bool have_max_objects)
-{
-  switch (opt_cmd) {
-    case OPT_QUOTA_ENABLE:
-    case OPT_GLOBAL_QUOTA_ENABLE:
-      quota.enabled = true;
+  int ret = parse_commandline_parameters(args, user_id, tenant, access_key, subuser, secret_key, user_email,user_op,
```

Comment:
I don't like this, too many parameters.
I suggest starting by parsing the command, than for each command parse the parameter for it.
This way we won't have it in main.